server/channels/api4/channel_common_teams_test.go

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package api4
+
+import (
+	"context"
+	"testing"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetDirectOrGroupMessageMembersCommonTeams(t *testing.T) {
+	mainHelper.Parallel(t)
+
+	th := Setup(t).InitBasic(t)
+	client := th.Client
+
+	t.Run("requires authentication", func(t *testing.T) {
```

Comment:
Are you testing `APISessionRequired` here? Feels like this should've been a static check. I mean do we need to test this over and over again for all APIs?

Reply:
That's a good question. Do we have a pattern for doing that kind of static check so we can avoid the repetitive test? I suppose I'm not testing `APISessionRequired` so much as I'm testing that the definition of this endpoint includes `APISessionRequired`.